test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -187,7 +189,7 @@ public void CreateDisplayMetadata_DisplayAttribute_NameFromResources()
         public void CreateDisplayMetadata_DisplayAttribute_DescriptionFromResources()
```

Comment:
Could turn this test and the similar ones for `Name` and `Prompt` into `Theories` where the `IStringLocalizerFactory` is passed in (`null` or non-`null`) the factory should either not be called (if you optimize `Create()` calls -- again, that's minor) or only called once (and then the `IStringLocalizer` would never be used).